### PR TITLE
Add matio entry for RHEL

### PR DESCRIPTION
**Pull request by wxmerkt**
Cf. https://packages.fedoraproject.org/pkgs/matio/matio/

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -6657,6 +6657,7 @@ matio:
   debian: [libmatio-dev]
   fedora: [matio-devel]
   nixos: [matio]
+  rhel: [matio]
```
Comment (cottsay) [inline rosdep/base.yaml:6660]: Two suggestions:
1. This isn't available for RHEL 9 yet, so let's just add it for RHEL 8 (which is probably where you want it anyway)
2. The other packages appear to get development headers, so we should get them here too.

```suggestion
  rhel:
    '*': null
    '8': [matio-devel]
```

**Review by cottsay — CHANGES_REQUESTED** [review]
Thanks for adding a rule for RHEL!

File: rosdep/base.yaml (hunk 1 above)
Comment (wxmerkt) [reply rosdep/base.yaml]: Awesome, thank you for the swift review and suggestions

**Review by cottsay — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Comment by wxmerkt** [comment]
I'd like to revisit this to update packages for Rolling & Iron: I cannot find matio for RHEL9. I may have missed those though, in those cases what is the proposed step to make packages that were released for Rolling on RHEL8 (as with Humble) to update for Rolling on RHEL9 (as with Iron)?

**Comment by cottsay** [comment]
It isn't available for RHEL 9 right now. I'll request that the package be built, but for now, you'll have to proceed without RHEL.

**Comment by wxmerkt** [comment]
Thank you - so I will run the release and acknowledge the prompt that we are absolutely certain it's not available

**Comment by cottsay** [comment]
> run the release and acknowledge the prompt that we are absolutely certain it's not available

Precisely, thanks for confirming.

**Pull request merged**